org.spoofax.terms/src/org/spoofax/interpreter/terms/InternalIStrategoArrayList.java

```diff
@@ -0,0 +1,23 @@
+package org.spoofax.interpreter.terms;
+
+import java.io.IOException;
+import org.spoofax.terms.StrategoArrayList;
+
+/**
+ * Internal interface used to share code between the {@link StrategoArrayList} and {@link StrategoArrayListTail}
```

Comment:
I'm confused. How is this sharing any code if there is no code? It's just an interface?

Reply:
Yeah, that's unclear. This interface is used by `IStrategoArrayList`, which provides the actual shared code

Reply:
It's still unclear what the purpose of `InternalIStrategoArrayList` is to me. Apart from one class that implements this interface, I don't see any references to it in this PR, who is the user of this interface?

Reply:
It's used by `IStrategoArrayList` only. I just separated the internal methods in a separate interface. I can include them in `IStrategoArrayList` instead if you want.